process.c

```diff
@@ -1267,35 +1303,82 @@ waitpid_no_SIGCHLD(struct waitpid_state *w)
         w->errnum = errno;
 }
 
-rb_pid_t
-rb_waitpid(rb_pid_t pid, int *st, int flags)
+// This is the root entry point for waiting on a pid.
```

Comment:
Could you mention it always returns a `Process::Status` instance?

Reply:
The scheduler docs should also mention the new callback and what it must return.

Reply:
Sure, it's just the last thing on my mind right now.

Reply:
I've added documentation.